### ISSUE TITLE
A11y : Cannot collapse query row via keyboard

### ISSUE THREAD

**Issue by zuchka** [grafana/grafana#46486]
**Steps to reproduce**:

Try to collapse the Explore query row using keyboard-only navigation

**Actual Result**:

Can’t collapse query row 

Also, there is a focus-issue here. See [the A11y notes](https://docs.google.com/spreadsheets/d/1e3sjrB3wjgQfttK5ANNkac5wmDwLszhgW0fO1kWT2LI/edit#gid=0):

> Same tab, when you copy the query or a link to it, the focus is reset

**Expected Result**

Repeatable UX for those not using a pointer device

**Relevant WCAG Criteria:**

[2.1.1 Keyboard - Level A](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0&showtechniques=335%2C332#keyboard)

**Environment**:
- Grafana version: 8.4.3.
- Data source type & version:
- User OS & Browser:
- Others:

**Comment by L-M-K-B** [comment]
> Also, there is a focus-issue here. See [the A11y notes](https://docs.google.com/spreadsheets/d/1e3sjrB3wjgQfttK5ANNkac5wmDwLszhgW0fO1kWT2LI/edit#gid=0):
> > Same tab, when you copy the query or a link to it, the focus is reset

After clarifying with [#wg-a11y](https://raintank-corp.slack.com/archives/C030MMKLH2M/p1660748234705599?thread_ts=1660747738.120919&cid=C030MMKLH2M) this relates to issue #46477.
